src/PointwiseFunctions/AnalyticData/Xcts/Binary.hpp

```diff
@@ -0,0 +1,350 @@
+    static constexpr Options::String help =
+        "The object placed on the negative x-axis";
+    using type = std::unique_ptr<IsolatedObjectBase>;
+  };
+  struct ObjectB {
+    static constexpr Options::String help =
+        "The object placed on the positive x-axis";
+    using type = std::unique_ptr<IsolatedObjectBase>;
+  };
+  struct AngularVelocity {
+    static constexpr Options::String help = "Orbital angular velocity";
+    using type = double;
+  };
+  struct FalloffWidths {
+    static constexpr Options::String help =
+        "The widths for the window functions around the two objects, or 'None' "
+        "to disable the Gaussian falloff.";
+    using type = Options::Auto<std::array<double, 2>, Options::AutoLabel::None>;
+  };
+  using options =
+      tmpl::list<XCoords, ObjectA, ObjectB, AngularVelocity, FalloffWidths>;
+  static constexpr Options::String help =
+      "Binary compact-object data in general relativity, constructed from "
+      "superpositions of two isolated objects.";
+
+  Binary() = default;
+  Binary(const Binary&) = delete;
+  Binary& operator=(const Binary&) = delete;
+  Binary(Binary&&) = default;
+  Binary& operator=(Binary&&) = default;
+  ~Binary() = default;
+
+  Binary(std::array<double, 2> xcoords,
+         std::unique_ptr<IsolatedObjectBase> object_a,
+         std::unique_ptr<IsolatedObjectBase> object_b, double angular_velocity,
+         std::optional<std::array<double, 2>> falloff_widths) noexcept
+      : xcoords_(xcoords),
+        superposed_objects_({std::move(object_a), std::move(object_b)}),
+        angular_velocity_(angular_velocity),
+        falloff_widths_(falloff_widths) {}
+
+  explicit Binary(CkMigrateMessage* m) noexcept : Base(m) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(Binary);
+
+  template <typename DataType, typename... RequestedTags>
+  tuples::TaggedTuple<RequestedTags...> variables(
+      const tnsr::I<DataType, 3, Frame::Inertial>& x,
+      tmpl::list<RequestedTags...> /*meta*/) const noexcept {
+    return variables_impl<DataType>(x, std::nullopt, std::nullopt,
+                                    tmpl::list<RequestedTags...>{});
+  }
+  template <typename... RequestedTags>
+  tuples::TaggedTuple<RequestedTags...> variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x, const Mesh<3>& mesh,
+      const InverseJacobian<DataVector, 3, Frame::Logical, Frame::Inertial>&
+          inv_jacobian,
+      tmpl::list<RequestedTags...> /*meta*/) const noexcept {
+    return variables_impl(x, mesh, inv_jacobian,
+                          tmpl::list<RequestedTags...>{});
+  }
+
+  // NOLINTNEXTLINE
+  void pup(PUP::er& p) noexcept override {
+    Base::pup(p);
+    p | xcoords_;
+    p | superposed_objects_;
+    p | angular_velocity_;
+    p | falloff_widths_;
+  }
+
+  std::array<double, 2> x_coords() const noexcept { return xcoords_; }
+  std::array<std::unique_ptr<IsolatedObjectBase>, 2> superposed_objects()
```

Comment:
I'm confused about what happens with the return type here -- does it try to copy the `unique_ptr`s?
I can't find anywhere this function is used -- are these accessor functions necessary? If they're needed, then it seems like this function needs to do some kind of `get_clone` trick or to return the contents by const reference.

Reply:
Good catch, I switched these accessors to return const-refs and added a small test.